sgp/Types.h

```diff
@@ -24,6 +24,7 @@
 #define endof(a) ((a) + lengthof(a))
 
 
+#define ABS(a) (UINT16)abs((int)a)
```

Comment:
Should be (int)(a) or simply int(a) so the result is cast, not only the first argument in non-trivial expressions.

And while the change seems automated, more whitespace errors crept up.